src/imagecaption/imagecaptionengine.js

```diff
@@ -201,42 +144,22 @@ function insertMissingModelCaptionElement( evt, changeType, data, batch ) {
 		}
 	}
 }
-
-// Returns function that should be executed when model to view conversion is made. It checks if insertion is placed
-// inside model caption and makes sure that corresponding view element exists.
-//
-// @private
-// @param {function} creator Function that returns view caption element.
-// @param {module:engine/conversion/mapper~Mapper} mapper
-// @return {function}
-function insertMissingViewCaptionElement( creator, mapper ) {
-	return ( evt, data ) => {
-		if ( isInsideCaption( data.item ) ) {
-			const modelCaption = data.item.parent;
-			const modelImage = modelCaption.parent;
-
-			const viewImage = mapper.toViewElement( modelImage );
-			let viewCaption =  mapper.toViewElement( modelCaption );
-
-			// Image should be already converted to the view.
-			if ( viewImage && !viewCaption ) {
-				viewCaption = creator();
-				insertViewCaptionAndBind( viewCaption, modelCaption, viewImage, mapper );
-			}
-		}
-	};
-}
-
 // Creates a converter that converts image caption model element to view element.
 //
 // @private
 // @param {Function|module:engine/view/element~Element} elementCreator
+// @param {Boolean} [hide=true] When set to `false` view element will not be inserted when it's empty.
 // @return {Function}
-function captionModelToView( elementCreator ) {
+function captionModelToView( elementCreator, hide = true ) {
 	return ( evt, data, consumable, conversionApi ) => {
 		const captionElement = data.item;
 
-		if ( isImage( captionElement.parent ) && ( captionElement.childCount > 0 ) ) {
+		// Return if element shouldn't be present when empty.
+		if ( !captionElement.childCount && !hide ) {
+			return;
```

Comment:
I think it would be better if this part of code is in `if ( isImage( ... ) ) { ... }` so it won't ever affect other converters if we have different type of captions.